types/jscodeshift/src/core.d.ts

```diff
@@ -0,0 +1,363 @@
+    type Parser = recast.Parser;
+
+    interface Filters {
+        JSXElement: JSXElement.FilterMethods;
+        VariableDeclarator: VariableDeclarator.FilterMethods;
+    }
+
+    interface Mappings {
+        JSXElement: JSXElement.MappingMethods;
+    }
+
+    interface Plugin {
+        (core: Core): void;
+    }
+
+    interface FileInfo {
+        /** The absolute path to the current file. */
+        path: string;
+        /** The source code of the current file. */
+        source: string;
+    }
+
+    interface Stats {
+        /**
+         * Helper function to collect data during --dry runs.
+         * This function keeps a counter for how often it was called with a specific argument.
+         * The result is shown in the console. Useful for finding out how many files match a criterion.
+         */
+        (name: string, quantity?: number): void;
+    }
+
+    type ASTPath<N = ASTNode> = recast.NodePath<N, N>;
+
+    interface Core {
+        (source: string, options?: Options): Collection.Collection<any>;
+        (source: ASTNode | ASTNode[] | ASTPath | ASTPath[]): Collection.Collection<any>;
+
+        registerMethods: typeof Collection.registerMethods;
+
+        types: typeof recast.default.types;
+
+        match(
+            path: ASTNode | ASTPath,
+            filter: ((path: ASTNode) => boolean) | ASTNode
+        ): boolean;
+
+        /** template, bound to default parser */
+        template: template.Template;
+
+        filters: Filters;
+
+        mappings: Mappings;
+
+        /**
+         * Utility function for registering plugins.
+         *
+         * Plugins are simple functions that are passed the core jscodeshift instance.
+         * They should extend jscodeshift by calling `registerMethods`, etc.
+         * This method guards against repeated registrations (the plugin callback will only be called once).
+         */
+        use(plugin: Plugin): void;
+
+        /**
+         * Returns a version of the core jscodeshift function "bound" to a specific
+         * parser.
+         */
+        withParser(parser: string | Parser): JSCodeshift;
+    }
+
+    type JSCodeshift = Core & recast.NamedTypes & recast.Builders;
+
+    interface API {
+        j: JSCodeshift;
+        jscodeshift: JSCodeshift;
+        stats: Stats;
+        report: (msg: string) => void;
+    }
+
+    interface Options {
+        [option: string]: any;
+    }
+
+    interface Transform {
+        /**
+         * If a string is returned and it is different from passed source, the transform is considered to be successful.
+         * If a string is returned but it's the same as the source, the transform is considered to be unsuccessful.
+         * If nothing is returned, the file is not supposed to be transformed (which is ok).
+         */
+        (file: FileInfo, api: API, options: Options): string | null | undefined | void;
+    }
+
+    type ASTNode = nodes.ASTNode;
+
+    type AnyTypeAnnotation = nodes.AnyTypeAnnotation;
```

Comment:
Before, I had used `export * from "ast-types/gen/nodes";` for this, but I couldn't get that working after changing this module to use `export =`. Is there a better way to do this?